### PR TITLE
Upgrade Doctrine

### PR DESCRIPTION
**Pull request by mabumusa1**
| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 

#### Description:

This step upgrade the required packages of doctrine, a follow up PR will come to fix `composer.json` versions and upgrade to the next version

**CS Fixer is failing because I upgraded the package, once we get this PR reviewed I will run it, I did not run it so the files to be reviewed is minimal**

#### Steps to test this PR:

1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Use the system in different parts including emails, reports, and contacts
3. Make sure it works

### CODE REVIEW

**Comment by mabumusa1** [comment]
After Commit cc780f852b84b0d12c69471b424414ffe169b3d2, these are all the packages that got updated

````
| Prod Packages                       | Operation | Base    | Target  |
|-------------------------------------|-----------|---------|---------|
| composer/xdebug-handler             | Upgraded  | 2.0.5   | 3.0.3   |
| doctrine/annotations                | Upgraded  | 1.14.3  | 2.0.1   |
| doctrine/common                     | Upgraded  | 3.4.0   | 3.4.3   |
| doctrine/data-fixtures              | Upgraded  | 1.4.4   | 1.6.6   |
| doctrine/dbal                       | Upgraded  | 2.13.9  | 3.6.2   |
| doctrine/doctrine-bundle            | Upgraded  | 2.5.7   | 2.9.1   |
| doctrine/doctrine-fixtures-bundle   | Upgraded  | 3.4.2   | 3.4.4   |
| doctrine/doctrine-migrations-bundle | Upgraded  | 2.2.3   | 3.2.2   |
| doctrine/event-manager              | Upgraded  | 1.1.2   | 1.2.0   |
| doctrine/inflector                  | Upgraded  | 2.0.4   | 2.0.6   |
| doctrine/instantiator               | Upgraded  | 1.4.1   | 1.5.0   |
| doctrine/lexer                      | Upgraded  | 1.2.3   | 2.1.0   |
| doctrine/migrations                 | Upgraded  | 2.3.5   | 3.5.5   |
| doctrine/orm                        | Upgraded  | 2.13.1  | 2.15.1  |
| doctrine/persistence                | Upgraded  | 2.5.4   | 3.2.0   |
| egulias/email-validator             | Upgraded  | 3.2.1   | 3.2.5   |
| friendsofsymfony/rest-bundle        | Upgraded  | 3.3.0   | 3.5.0   |
| jms/serializer                      | Upgraded  | 3.18.1  | 3.23.0  |
| jms/serializer-bundle               | Upgraded  | 4.0.2   | 5.1.0   |
| noxlogic/ratelimit-bundle           | Upgraded  | v1.18.0 | v1.19.0 |
| phpstan/phpdoc-parser               | Upgraded  | 1.16.1  | 1.21.0  |
| psr/cache                           | Upgraded  | 1.0.1   | 2.0.0   |
| sensio/framework-extra-bundle       | Upgraded  | v6.2.8  | v6.2.10 |
| symfony/cache                       | Upgraded  | v5.4.22 | v5.4.23 |
| symfony/dependency-injection        | Upgraded  | v5.4.22 | v5.4.24 |
| symfony/error-handler               | Upgraded  | v5.4.21 | v5.4.24 |
| symfony/event-dispatcher-contracts  | Upgraded  | v2.5.2  | v3.0.2  |
| symfony/filesystem                  | Upgraded  | v5.4.21 | v5.4.23 |
| symfony/framework-bundle            | Upgraded  | v5.4.22 | v5.4.24 |
| symfony/http-foundation             | Upgraded  | v5.4.22 | v5.4.24 |
| symfony/http-kernel                 | Upgraded  | v5.4.22 | v5.4.24 |
| symfony/mime                        | Upgraded  | v5.4.21 | v6.0.19 |
| symfony/password-hasher             | Upgraded  | v5.4.21 | v6.0.19 |
| symfony/property-info               | Upgraded  | v5.4.22 | v6.0.19 |
| symfony/string                      | Upgraded  | v5.4.22 | v6.0.19 |
| symfony/translation                 | Upgraded  | v5.4.22 | v5.4.24 |
| symfony/var-exporter                | Upgraded  | v5.4.21 | v6.0.19 |

| Dev Packages                | Operation | Base   | Target  |
|-----------------------------|-----------|--------|---------|
| friendsofphp/php-cs-fixer   | Upgraded  | v3.4.0 | v3.14.1 |
| liip/functional-test-bundle | Upgraded  | 4.5.1  | 4.9.0   |
| liip/test-fixtures-bundle   | Upgraded  | 2.4.1  | 2.6.0   |
| phpdocumentor/type-resolver | Upgraded  | 1.7.0  | 1.7.1   |
| php-cs-fixer/diff           | Removed   | v2.0.2 | -       |
````

**Comment by mabumusa1** [comment]
Here are the blockers for this PR to go green

- [x] a rewrite to this [function](https://github.com/mautic/mautic/blob/87a71ea110e033202938a82da9b5adec9dcc23cf/app/bundles/ReportBundle/Builder/MauticReportBuilder.php#L371) , right now [this line](https://github.com/mautic/mautic/blob/87a71ea110e033202938a82da9b5adec9dcc23cf/app/bundles/ReportBundle/Builder/MauticReportBuilder.php#L375) is adding an extra `()` to the query which fails the test 

- [x]  `SimplePaginatorTest` is failing due the introduction of `Middlewares` which can not be changed during runtime, they need to be added during compile, so it is a good idea to add them to `config_test.php`. [Here is more details about it ](https://github.com/doctrine/dbal/issues/5366#issuecomment-1110254314)

Based on @fedys advise, I removed the queries test at the end in `SimplePaginatorTest`

- [x]  Database should be hard coded into config as [mentioned here](https://symfony.com/doc/current/reference/configuration/doctrine.html#doctrine-dbal-configuration) 

- [x] There is this error, I can not find a way to mock it since it is final
```1) Mautic\LeadBundle\Tests\Field\CustomFieldColumnTest::testCustomFieldLimit
PHPUnit\Framework\MockObject\MethodCannotBeConfiguredException: Trying to configure method "getCode" which cannot be configured because it does not exist, has not been specified, is final, or is static

/home/runner/work/mautic/mautic/app/bundles/LeadBundle/Tests/Field/CustomFieldColumnTest.php:152
```



**Blockers to upgrade to a higher version**

- [ ] Some files inside `app/bundles/LeadBundle/Segment/Query/` are a copy and past of `doctrine`, they will break because doctrine has changed, some of the functions needs to be implemented as as DQL or used from this pacakge `beberlei/doctrineextensions` like `regexp`

**Comment by escopecz** [comment]
@mabumusa1 please check https://github.com/mautic/mautic/pull/12403 for the first point

**Comment by mabumusa1** [comment]
All of these packages got updated 

````
| Prod Packages                       | Operation | Base    | Target  |
|-------------------------------------|-----------|---------|---------|
| composer/xdebug-handler             | Upgraded  | 2.0.5   | 3.0.3   |
| doctrine/annotations                | Upgraded  | 1.14.3  | 2.0.1   |
| doctrine/common                     | Upgraded  | 3.4.0   | 3.4.3   |
| doctrine/data-fixtures              | Upgraded  | 1.4.4   | 1.6.6   |
| doctrine/dbal                       | Upgraded  | 2.13.9  | 3.6.2   |
| doctrine/doctrine-bundle            | Upgraded  | 2.5.7   | 2.9.1   |
| doctrine/doctrine-fixtures-bundle   | Upgraded  | 3.4.2   | 3.4.4   |
| doctrine/doctrine-migrations-bundle | Upgraded  | 2.2.3   | 3.2.2   |
| doctrine/event-manager              | Upgraded  | 1.1.2   | 1.2.0   |
| doctrine/inflector                  | Upgraded  | 2.0.4   | 2.0.6   |
| doctrine/instantiator               | Upgraded  | 1.4.1   | 1.5.0   |
| doctrine/lexer                      | Upgraded  | 1.2.3   | 2.1.0   |
| doctrine/migrations                 | Upgraded  | 2.3.5   | 3.5.5   |
| doctrine/orm                        | Upgraded  | 2.13.1  | 2.15.1  |
| doctrine/persistence                | Upgraded  | 2.5.4   | 3.2.0   |
| egulias/email-validator             | Upgraded  | 3.2.1   | 3.2.5   |
| friendsofsymfony/rest-bundle        | Upgraded  | 3.3.0   | 3.5.0   |
| jms/serializer                      | Upgraded  | 3.18.1  | 3.23.0  |
| jms/serializer-bundle               | Upgraded  | 4.0.2   | 5.1.0   |
| noxlogic/ratelimit-bundle           | Upgraded  | v1.18.0 | v1.19.0 |
| phpstan/phpdoc-parser               | Upgraded  | 1.16.1  | 1.21.0  |
| psr/cache                           | Upgraded  | 1.0.1   | 2.0.0   |
| sensio/framework-extra-bundle       | Upgraded  | v6.2.8  | v6.2.10 |
| symfony/cache                       | Upgraded  | v5.4.22 | v5.4.23 |
| symfony/dependency-injection        | Upgraded  | v5.4.22 | v5.4.24 |
| symfony/error-handler               | Upgraded  | v5.4.21 | v5.4.24 |
| symfony/event-dispatcher-contracts  | Upgraded  | v2.5.2  | v3.0.2  |
| symfony/filesystem                  | Upgraded  | v5.4.21 | v5.4.23 |
| symfony/framework-bundle            | Upgraded  | v5.4.22 | v5.4.24 |
| symfony/http-foundation             | Upgraded  | v5.4.22 | v5.4.24 |
| symfony/http-kernel                 | Upgraded  | v5.4.22 | v5.4.24 |
| symfony/mime                        | Upgraded  | v5.4.21 | v6.0.19 |
| symfony/password-hasher             | Upgraded  | v5.4.21 | v6.0.19 |
| symfony/property-info               | Upgraded  | v5.4.22 | v6.0.19 |
| symfony/string                      | Upgraded  | v5.4.22 | v6.0.19 |
| symfony/translation                 | Upgraded  | v5.4.22 | v5.4.24 |
| symfony/var-exporter                | Upgraded  | v5.4.21 | v6.0.19 |

| Dev Packages                | Operation | Base    | Target  |
|-----------------------------|-----------|---------|---------|
| friendsofphp/php-cs-fixer   | Upgraded  | v3.4.0  | v3.14.1 |
| liip/functional-test-bundle | Upgraded  | 4.5.1   | 4.9.0   |
| liip/test-fixtures-bundle   | Upgraded  | 2.4.1   | 2.6.0   |
| phpdocumentor/type-resolver | Upgraded  | 1.7.0   | 1.7.1   |
| phpstan/phpstan-doctrine    | Upgraded  | 1.3.18  | 1.3.40  |
| symfony/phpunit-bridge      | Upgraded  | v5.1.11 | v5.4.23 |
| php-cs-fixer/diff           | Removed   | v2.0.2  | -       |


````

**Comment by escopecz** [comment]
I run the API Library test suite against this PR and there are 2 failures: https://github.com/mautic/api-library/actions/runs/5123472827/jobs/9214019270

File: app/bundles/CoreBundle/Tests/Functional/Doctrine/Paginator/SimplePaginatorTest.php [hunk 1]
```diff
@@ -18,18 +16,10 @@ class SimplePaginatorTest extends MauticMysqlTestCase
...
     protected function setUp(): void
     {
         parent::setUp();
-
-        $logger = self::$container->get('doctrine.dbal.logger.profiling.default');
-        assert($logger instanceof DebugStack);
-        $logger->queries      = [];
-        $logger->currentQuery = 0;
-
-        $this->logger = $logger;
```
Comment (escopecz) [inline app/bundles/CoreBundle/Tests/Functional/Doctrine/Paginator/SimplePaginatorTest.php]: @fedys could you please see if there is a way to do this in the new Doctrine version? Me and @mabumusa1 failed to  find a way.

Comment (fedys) [reply app/bundles/CoreBundle/Tests/Functional/Doctrine/Paginator/SimplePaginatorTest.php]: @escopecz I will take a look.

**Comment by mabumusa1** [comment]
> I run the API Library test suite against this PR and there are 2 failures: https://github.com/mautic/api-library/actions/runs/5123472827/jobs/9214019270

Fixed in commit 63d665986d0242a7359110502dc1932e3b431103, I tried to run the test suite but it failed on `citrix` https://github.com/mautic/api-library/actions/runs/5129793845/jobs/9227794537

**Comment by escopecz** [comment]
@mabumusa1 I chose to use your branch which is fixing the Citrix issue.

Screenshot 2023-05-31 at 11 15 39


The tests are now running in https://github.com/mautic/api-library/actions/runs/5131065142

**Comment by mabumusa1** [comment]
> @mabumusa1 I chose to use your branch which is fixing the Citrix issue.
> 
> Screenshot 2023-05-31 at 11 15 39
> 
> 
> The tests are now running in https://github.com/mautic/api-library/actions/runs/5131065142

Good tip, I think it works now as expected

File: app/bundles/CoreBundle/Tests/Functional/Doctrine/Paginator/SimplePaginatorTest.php (hunk 1 above)
Comment (escopecz) [reply app/bundles/CoreBundle/Tests/Functional/Doctrine/Paginator/SimplePaginatorTest.php]: I discussed this with Fedy. Asserting the queries are not that important in this test. The output is being asserted and that matters. We can get back to solving the query logging later but it should not block this PR and M5 release.

**Review by escopecz — DISMISSED** [review]
I cannot spot any issue with these changes. Thank you! 👍 

Although it would be better to update unrelated packages in a separate PR and focus this one on Doctrine. But since it's done and we'd have to do it anyway, let's run with this.

**Comment by mabumusa1** [comment]
The dependences have circular reference, it would be hard to do each package on a separate PR

**Comment by fedys** [comment]
@mabumusa1 I reverted 5564e65dc030926eb15ddf58fbd742bcfaef9b8f and 64d3647ae1ed2bbad9f6d3b02b0e06a750f03ea8 as they are not needed and aren't valid. The query profiler was not working for me because of my envs being set improperly.

**Comment by mabumusa1** [comment]
> @mabumusa1 I reverted [5564e65](https://github.com/mautic/mautic/commit/5564e65dc030926eb15ddf58fbd742bcfaef9b8f) and [64d3647](https://github.com/mautic/mautic/commit/64d3647ae1ed2bbad9f6d3b02b0e06a750f03ea8) as they are not needed and aren't valid. The query profiler was not working for me because of my envs being set improperly.

Okay, thanks. Do you think this is ready to go?

**Comment by fedys** [comment]
@mabumusa1 @escopecz I found a way to bring back the query assertions in https://github.com/mautic/mautic/pull/12396/commits/f7debd8cb29156ff74e14df0be29a98ad67cd676.

**Review by fedys — APPROVED** [review]
Thanks @mabumusa1! The PR is good to go.

File: app/bundles/CoreBundle/Tests/Functional/Doctrine/Paginator/SimplePaginatorTest.php (hunk 1 above)
Comment (fedys) [reply app/bundles/CoreBundle/Tests/Functional/Doctrine/Paginator/SimplePaginatorTest.php]: Resolved in https://github.com/mautic/mautic/commit/f7debd8cb29156ff74e14df0be29a98ad67cd676

**Review by escopecz — APPROVED** [review]
Thank you both! 👍

**Pull request merged**